firedrake/interpolation.py

```diff
@@ -472,7 +472,11 @@ def __init__(
             raise NotImplementedError("freeze_expr not implemented")
         if bcs:
             raise NotImplementedError("bcs not implemented")
-        if V.ufl_element().mapping() != "identity":
+
+        target_element = V.ufl_element()
+        if not ((isinstance(target_element, finat.ufl.MixedElement)
+                 and all(sub.mapping() == "identity" for sub in target_element.sub_elements))
+                or target_element.mapping() == "identity"):
```

Comment:
Didn't you recently add a utility function for checking for only point evaluation nodes? Might that not be better here?